### DISCUSSION TITLE
Multiple instance of cog

### DISCUSSION THREAD

**Discussion by GobGeeklord** [Igalia/cog#472] [Q&A] [answered]
Hi there,

I'm not sure if this question is duplicated with multiple-views, but what i want to know is possible to run multiple cog instances with completely separated apps?
The purpose is not to block app2 (render?) when app1 is blocked executing calls

Example:
![image](https://user-images.githubusercontent.com/85292943/173819875-d355241d-a5b2-452d-b14a-6aef202efcab.png)

Is this scenario even possible?

**Comment by ventosus** [comment, marked as answer]
I have a similar question. I have weston configured in kiosk mode for two monitors and would like to run two instances of cog. Whenever I start the second cog instance, the first is blacked out (but still seems to run). I Have the feeling, there may be a name clash with dbus, but I am not sure.

I use the gapplication ids and the seem to work, e.g. the browser start up on the correct display, but the first disappears when the second one starts up, but reappears, when the second is stopped.

# weston.ini

    [core]
    shell=kiosk-shell.so
    
    [output]
    name=DP-1
    mode=28.10  1280 1283 1288 1292  720 721 723 725 +hsync +vsync
    app-ids=com.example.Left
    
    [output]
    name=eDP-1
    mode=28.10  1280 1283 1288 1292  720 721 723 725 +hsync +vsync
    app-ids=com.example.Right


# cog left

    cog --gapplication-app-id=com.example.Left http://left.com


# cog right

    cog --gapplication-app-id=com.example.Right http://right.com

**Reply by GobGeeklord** [reply]
Hi @ventosus ,
This is exactly what i was looking for! But the question is, the cog "spawn" seems to be very random (window position), is there any way to control this  from a weston perspective? (window size, position, etc?)

**Reply by ventosus** [reply]
I wouldn't know, I use the kiosk shell exclusively, it's always fullscreen

**Comment by ventosus** [comment]
When run in weston's desktop-shell.so, both cog browsers are visible, but the desktop shell is really inconvenient for a kiosk system.

**Comment by ventosus** [comment]
This was apparently a weston bug and got fixed in weston 10.0.1.

**Comment by commodo** [comment]
> This was apparently a weston bug and got fixed in weston 10.0.1.

thanks for that hint; it helped me (at least)  :)

**Comment by aperezdc** [comment]
Using different application identifiers, as suggested by @ventosus, is expected to work from Cog's point of view. This uses two different `cog` processes.

Note that you can also use `cogctl --appid=` to indicate which instance's remote control interface to use. 

The support for multiple views (see #77) will allow using more than one web view per `cog` process.
